test/battle/move_effect/bestow.c

```diff
@@ -1,10 +1,132 @@
 #include "global.h"
 #include "test/battle.h"
 
-TO_DO_BATTLE_TEST("Bestow transfers its held item to the target");
-TO_DO_BATTLE_TEST("Bestow fails if the user has no held item");
-TO_DO_BATTLE_TEST("Bestow fails if the target already has a held item");
-TO_DO_BATTLE_TEST("Bestow fails if the target is behind a Substitute");
-TO_DO_BATTLE_TEST("Bestow fails if the user is holding Mail");
-TO_DO_BATTLE_TEST("Bestow fails if the user's held item changes its form");
-TO_DO_BATTLE_TEST("Bestow fails if the user's held item is a Z-Crystal");
+SINGLE_BATTLE_TEST("Bestow transfers its held item to the target")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_SITRUS_BERRY); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_BESTOW); }
+    } THEN {
+        EXPECT(player->item == ITEM_NONE);
+        EXPECT(opponent->item == ITEM_SITRUS_BERRY);
+    }
+}
+
+SINGLE_BATTLE_TEST("Bestow fails if the user has no held item")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_BESTOW); }
+    } SCENE {
+        MESSAGE("But it failed!");
+    } THEN {
+        EXPECT(player->item == ITEM_NONE);
+        EXPECT(opponent->item == ITEM_NONE);
+    }
+}
+
+SINGLE_BATTLE_TEST("Bestow fails if the target already has a held item")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_SITRUS_BERRY); }
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_LUM_BERRY); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_BESTOW); }
+    } SCENE {
+        MESSAGE("But it failed!");
+    } THEN {
+        EXPECT(player->item == ITEM_SITRUS_BERRY);
+        EXPECT(opponent->item == ITEM_LUM_BERRY);
+    }
+}
+
+#include "mail.h"
+SINGLE_BATTLE_TEST("Bestow fails if the user is holding Mail")
+{
+    KNOWN_FAILING;
+
+    ASSUME(ItemIsMail(ITEM_ORANGE_MAIL));
+
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_ORANGE_MAIL); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_BESTOW); }
+    } SCENE {
+        MESSAGE("But it failed!");
+    } THEN {
+        EXPECT(player->item == ITEM_ORANGE_MAIL);
+        EXPECT(opponent->item == ITEM_NONE);
+    }
+}
+
+SINGLE_BATTLE_TEST("Bestow fails if the user's held item is a Mega Stone")
+{
+    GIVEN {
+        PLAYER(SPECIES_BLAZIKEN) { Item(ITEM_BLAZIKENITE); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_BESTOW); }
+    } SCENE {
+        MESSAGE("But it failed!");
+    } THEN {
+        EXPECT(player->item == ITEM_BLAZIKENITE);
+        EXPECT(opponent->item == ITEM_NONE);
+    }
+}
+
+SINGLE_BATTLE_TEST("Bestow fails if the user's held item is a Z-Crystal")
+{
+    ASSUME(ItemId_GetHoldEffect(ITEM_FIGHTINIUM_Z) == HOLD_EFFECT_Z_CRYSTAL);
+
+    GIVEN {
```

Comment:
```suggestion
    GIVEN {
        ASSUME(ItemId_GetHoldEffect(ITEM_FIGHTINIUM_Z) == HOLD_EFFECT_Z_CRYSTAL);
```
`ASSUME` outside of `GIVEN`